frontend/src/actions/public_file.js

```diff
@@ -0,0 +1,209 @@
+import config from "../app.config";
+import { V2 } from "../openapi";
+import { handleErrors } from "./common";
+
+export const RECEIVE_PUBLIC_FILE_METADATA = "RECEIVE_PUBLIC_FILE_METADATA";
+
+export function fetchPublicFileMetadata(fileId, version) {
+	return (dispatch) => {
+		return V2.PublicFilesService.getFileMetadataApiV2PublicFilesFileIdMetadataGet(
+			fileId,
+			version,
+			false
+		)
+			.then((json) => {
+				dispatch({
+					type: RECEIVE_PUBLIC_FILE_METADATA,
+					publicFileMetadataList: json,
+					receivedAt: Date.now(),
+				});
+			})
+			.catch((reason) => {
+				dispatch(handleErrors(reason, fetchPublicFileMetadata(fileId, version)));
+			});
+	};
+}
+
+export const RECEIVE_PUBLIC_FILE_EXTRACTED_METADATA =
+	"RECEIVE_PUBLIC_FILE_EXTRACTED_METADATA";
+
+export function fetchPublicFileExtractedMetadata(id) {
+	const url = `${config.hostname}/api/v2/public/files/${id}/metadata`;
+	return (dispatch) => {
+		return fetch(url, { mode: "cors"})
+			.then((response) => {
+				if (response.status === 200) {
+					response.json().then((json) => {
+						dispatch({
+							type: RECEIVE_PUBLIC_FILE_EXTRACTED_METADATA,
+							publicExtractedMetadata: json,
+							receivedAt: Date.now(),
+						});
+					});
+				} else {
+					dispatch(handleErrors(response, fetchPublicFileExtractedMetadata(id)));
+				}
+			})
+			.catch((reason) => {
+				dispatch(handleErrors(reason, fetchPublicFileExtractedMetadata(id)));
+			});
+	};
+}
+
+export const RECEIVE_PUBLIC_FILE_SUMMARY = "RECEIVE_PUBLIC_FILE_SUMMARY";
+
+export function fetchPublicFileSummary(id) {
+	return (dispatch) => {
+		return V2.PublicFilesService.getFileSummaryApiV2PublicFilesFileIdSummaryGet(id)
+			.then((json) => {
+				dispatch({
+					type: RECEIVE_PUBLIC_FILE_SUMMARY,
+					publicFileSummary: json,
+					receivedAt: Date.now(),
+				});
+			})
+			.catch((reason) => {
+				dispatch(handleErrors(reason, fetchPublicFileSummary(id)));
+			});
+	};
+}
+
+export const RECEIVE_PUBLIC_FILE_METADATA_JSONLD = "RECEIVE_PUBLIC_FILE_METADATA_JSONLD";
+
+export function fetchPublicFileMetadataJsonld(id) {
+	const url = `${config.hostname}/public/files/${id}/metadata.jsonld`;
```

Comment:
Hmm do we even have a jsonld endpoint? you probably can leave this out

Reply:
removed.